apps/admin/backend/src/app.ts

```diff
@@ -898,6 +905,35 @@ function buildApi({
       });
     },
 
+    async printWriteInAdjudicationReport(): Promise<void> {
```

Comment:
I'm encountering a general testing dilemma with rendering PDFs on the backend. Above this method, there's the `getElectionWriteInSummary` method. Its purpose is to provide the data for building a WIA report. But now the frontend doesn't need it - yet I'm not removing it.

It's nice to test at the app-level, but if the app only exposes PDFs, it's trickier to test than data blobs. There are some tests in `app.write_ins.test.ts` that depend heavily on this endpoint. But... the app doesn't use it so there's a risk of drift between the actual interface and the testing interface. I can think of a few ways to deal with them, none too bad or too good:

1. Let the endpoint stick around and use it for testing, the risk of divergence is small.
2. Develop means to read the text content of PDFs and parse it, so that we can actually extract the information from the PDFs. This is annoying and fragile testing, but it would do the best job of testing the actual output of the app. I explored this a bit, leveraging the Linux `pdftotext` command-line call. It seemed possible, but like a good deal of work parsing the text output.
3. Test this stuff a level up (or down?), not at the `app` level. I like the sound of that, but I think these sorts of tests can be finicky in a different way, because between the `store` and the `app` we don't have a clear intermediate layer defined to test at. Tests would involve sort of cobbling together helpers, store functions, and some logic that needs to be extracted from `app.ts`.

Suggestions or other ideas? I would argue it's fine to leave it as-is for now, but I want to think through the problem.

Reply:
Could you call `this.getElectionWriteInSummary()` in this function to ensure you're using the same data?

Reply:
That's a cool idea!